tests/build-std/main.rs

```diff
@@ -227,3 +227,50 @@ fn custom_test_framework() {
         .build_std_arg("core")
         .run();
 }
+
+/// like cross-custom but uses per-package-target instead
+#[cargo_test(build_std_real)]
```

Comment:
Can you say why this needs to be a real build-std test? I would prefer to avoid those if possible. The mock tests should be able to use `.arg("target").arg(alternate())` and use the `alternate` function to specify a target different from the host (and not using a JSON target).